lib/geo.js

```diff
@@ -10,23 +10,31 @@ var assert = require('assert');
  */
 
 exports.nearFilter = function nearFilter(where) {
+  var keyPath = [];
   var result = false;
 
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
-      }
-    });
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      Object.keys(node).forEach(function(key) {
+        var ex = node[key];
+        keyPath.push(key);
+
+        if (ex && ex.near) {
+          result = {
+            near: ex.near,
+            maxDistance: ex.maxDistance,
+            unit: ex.unit,
+            key: keyPath.slice(0),
```

Comment:
@jannyHou - I like that. I'd be happy to refactor it post merge if you want to assign an issue to me. It would also need to account for `or` right? I can do it in this PR, but ideally, if there are no other issues with it, it's be better for me to put that off to later since I have a project deadline kicking me in the face  :)

`key` is used by the connector's implementation of `buildNearFilter` to find the `near` property in the `where` filter - as far as I can tell, only the MongoDB connector uses this, so it shouldn't be a more widely breaking change.